src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/tests/Preview.test.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import React from 'react';
+import {observable} from 'mobx';
+import {mount, render, shallow} from 'enzyme';
+import ResourceStore from 'sulu-admin-bundle/stores/ResourceStore';
+import FormStore from 'sulu-admin-bundle/containers/Form/stores/FormStore';
+import Router from 'sulu-admin-bundle/services/Router';
+import Preview from '../Preview';
+
+jest.mock('debounce', () => jest.fn((value) => value));
+
+jest.mock('../stores/PreviewStore', () => jest.fn(function() {
+    this.start = jest.fn().mockReturnValue(Promise.resolve());
+    this.update = jest.fn().mockReturnValue(Promise.resolve());
+    this.updateContext = jest.fn().mockReturnValue(Promise.resolve());
+    this.stop = jest.fn().mockReturnValue(Promise.resolve());
+
+    this.renderRoute = '/render';
+}));
+
+jest.mock('sulu-admin-bundle/services/Requester', () => ({
+    get: jest.fn().mockImplementation((route: string) => new Promise((resolve) => {
+        if (route === '/start') {
+            resolve({token: '123-123-123'});
+        }
+    })),
+    post: jest.fn().mockReturnValue(Promise.resolve()),
+}));
+
+jest.mock('sulu-admin-bundle/containers/Form/stores/FormStore', () => jest.fn(function() {
+}));
+
+jest.mock('sulu-admin-bundle/stores/ResourceStore', () => jest.fn(function() {
+}));
+
+jest.mock('sulu-admin-bundle/services/Router', () => jest.fn(function(history) {
+    this.history = history;
+    this.attributes = {locale: 'de'};
+}));
+
+beforeEach(() => {
+    jest.resetModules();
+
+    Preview.mode = 'on_request';
+});
+
+test('Render correct preview', () => {
+    const resourceStore = new ResourceStore('pages', 1, {title: 'Test'});
+    const formStore = new FormStore(resourceStore);
+    const router = new Router({});
+
+    const component = shallow(<Preview formStore={formStore} router={router} />);
+
+    const startPromise = Promise.resolve();
+    const previewStore = component.instance().previewStore;
+    previewStore.start.mockReturnValue(startPromise);
+    previewStore.starting = false;
+
+    component.instance().handleStartClick();
+
+    startPromise.then(() => {
+        expect(component).toMatchSnapshot();
+    });
+});
+
+test('Render button to start preview', () => {
+    const resourceStore = new ResourceStore('pages', 1, {title: 'Test'});
+    const formStore = new FormStore(resourceStore);
+    const router = new Router({});
+
+    expect(render(<Preview formStore={formStore} router={router} />)).toMatchSnapshot();
+});
+
+test('React and update preview when data is changed', () => {
+    const resourceStore = new ResourceStore('pages', 1, {title: 'Test'});
+    const formStore = new FormStore(resourceStore);
+
+    // $FlowFixMe
```

Comment:
You can save these comments if you build the data and loading stuff a bit different, as e.g. in https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/tests/FormInspector.test.js#L7. Then you also don't pass the arguments from the `ResourceStore` for nothing :wink: